The-Chonky-Panda/Atelier
# Submit review

## Description

Added the ability to post new request for a product.
Changed characteristics rating display in review modal to only display applicable characteristics (i.e. what's shown in product breakdown)
Added a validation framework to validate user input. Currently the style for each individual field hasn't work yet, failed validation would only display a message.

## How to test

1. switch between products and verify that in the write new review modal, characteristic rating only displays characteristics that are shown in product breakdown
2. fill out the review modal with invalid inputs (e.g, empty input for required fields, review body is less than 50 characters, email is not in the format of an email), and verify that on click 'submit review', the modal remains open, and on top of the modal a message in red shows up.
<img width="727" alt="Screen Shot 2021-09-27 at 4 50 34 PM" src="https://user-images.githubusercontent.com/43324065/134991603-c13083e9-a19c-47f4-94e0-3ffa1fdfdfd9.png">
3. fill out the review modal with valid inputs, and verify that on click 'submit review', the modal closes and review list is updated.

## Notes


## Issue tracking

https://trello.com/c/RpIiO4L6
